app/utils/nonsipp/TaskListUtils.scala

```diff
@@ -387,23 +391,30 @@ object TaskListUtils {
     val (sharesDisposalsStatus, sharesDisposalsLinkUrl) =
       TaskListStatusUtils.getSharesDisposalsTaskListStatusWithLink(userAnswers, srn)
 
-    TaskListSectionViewModel(
-      s"$prefix.title",
-      TaskListItemViewModel(
-        LinkMessage(
-          messageKey(prefix, "sponsoringemployer.title", sharesStatus),
-          sharesLink
-        ),
-        sharesStatus
+    val sharesItem = TaskListItemViewModel(
+      LinkMessage(
+        messageKey(prefix, "sponsoringemployer.title", sharesStatus),
+        sharesLink
       ),
-      TaskListItemViewModel(
-        LinkMessage(
-          messageKey("nonsipp.tasklist.sharesdisposal", "title", sharesDisposalsStatus),
-          sharesDisposalsLinkUrl
-        ),
-        sharesDisposalsStatus
-      )
+      sharesStatus
     )
+
+    val sharesDisposalItem = TaskListItemViewModel(
+      LinkMessage(
+        messageKey("nonsipp.tasklist.sharesdisposal", "title", sharesDisposalsStatus),
+        sharesDisposalsLinkUrl
+      ),
+      sharesDisposalsStatus
+    )
+
+    val viewModelList = NonEmptyList
+      .fromList(userAnswers.get(SharesCompleted.all(srn)) match {
+        case Some(_) => List(sharesItem, sharesDisposalItem)
+        case None => List(sharesItem)
```

Comment:
Is this supposed to be only when DidSchemeHoldAnySharesPage is not  Some(true) ? As when `SharesCompleted` is Some(_) , there still can be some share disposals? Happy to discuss....